File: base64_and_file/src/main/java/com/phone/base64_and_file/adapter/Base64StrAdapter.java
package com.phone.base64_and_file.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.phone.base64_and_file.R;

import java.util.ArrayList;
import java.util.List;

public class Base64StrAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private static final String TAG = Base64StrAdapter.class.getSimpleName();
    public List<String> base64StrList = new ArrayList<>();
    private Context context;

    public Base64StrAdapter(Context context) {
        this.context = context;
    }

    public Base64StrAdapter(List<String> base64StrList, Context context) {
        this.base64StrList = base64StrList;
        this.context = context;
    }

    public void clearData() {
        notifyItemRangeRemoved(0, this.base64StrList.size());
        this.base64StrList.clear();
    }

    public void addData(List<String> base64StrList) {
        notifyItemRangeInserted(this.base64StrList.size(), base64StrList.size());
        this.base64StrList.addAll(base64StrList);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_base64_str, parent, false);
        return new BodyHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof BodyHolder) {
            BodyHolder bodyHolder = (BodyHolder) holder;
            bodyHolder.tevBase64Str.setText(base64StrList.get(position));
        }
    }

    @Override
    public int getItemCount() {
        return base64StrList.size();
    }

    private class BodyHolder extends RecyclerView.ViewHolder {

        private TextView tevBase64Str;

        public BodyHolder(@NonNull View itemView) {
            super(itemView);

            tevBase64Str = (TextView) itemView.findViewById(R.id.tev_base64_str);
        }
    }
}

File: base64_and_file/src/main/java/com/phone/base64_and_file/adapter/Base64StrAdapter.kt
package com.phone.base64_and_file.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.phone.base64_and_file.R

class Base64StrAdapter(val context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private val TAG = Base64StrAdapter::class.java.simpleName
    var base64StrList = mutableListOf<String>()

    fun clearData() {
        notifyItemRangeRemoved(0, base64StrList.size)
        notifyItemRangeChanged(0, base64StrList.size)
        base64StrList.clear()
    }

    fun addData(base64StrList: List<String>) {
        notifyItemRangeInserted(this.base64StrList.size, base64StrList.size)
        this.base64StrList.addAll(base64StrList)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view: View =
            LayoutInflater.from(context).inflate(R.layout.item_base64_str, parent, false)
        return BodyHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is BodyHolder) {
            holder.tevBase64Str.text = base64StrList[position]
        }
    }

    override fun getItemCount(): Int {
        return base64StrList.size
    }

    private class BodyHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tevBase64Str: TextView

        init {
            tevBase64Str = itemView.findViewById<View>(R.id.tev_base64_str) as TextView
        }
    }
}